### ISSUE TITLE
`Lspsaga diagnostic_jump_next` doesn't show current cursor diagnostic; instead it jumps to the next diagnostic

### ISSUE THREAD

**Issue by kohane27** [nvimdev/lspsaga.nvim#1467]
### Describe the bug

https://github.com/nvimdev/lspsaga.nvim/assets/57322459/334a773e-1e76-45a2-a155-8e0e88f2e370

Test with below:

```js
unction greet() {
  console.log("Hello World!");
}

greet();
```

1. Cursor is at `unction`
2. Run `Lspsaga diagnostic_jump_next`

Expect to show `Unknown keyword or identifier. Did you mean 'function'?`
Instead, it jumps to the next diagnostic and gives `';' expected. 1:17:1 tsserver 1005`

### Steps to reproduce

According to the [doc](https://nvimdev.github.io/lspsaga/diagnostic/):

> Invoke :Lspsaga diagnostic_jump_next and :Lspsaga diagnostic_jump_prev to jump around diagnostics. When the current cursor position has a diagnostic available, the commands will show the diagnostic in a nice float.

### Expected behavior

When the current cursor position has a diagnostic available, the commands will show the diagnostic in a nice float.

### Neovim version (nvim -v)

v0.10.0

### lspsaga commit

6f920cf

### Terminal name/version

kitty 0.35.1

**Comment by Kyren223** [comment]
@glepnir any estimate on when this will be fixed?
I want to be able to open that menu where my cursor is.
Alternatively, have another way to show this menu for the current cursor position (with a command for it)

**Comment by atusy** [comment]
@Kyren223 

Try

```
Lspsaga show_cursor_diagnostics
```

**Comment by Kyren223** [comment]
Yea that seems to work, thanks.
I think the original intent was for the jump_next to not jump if it's already on a diagnostic, but that might be undesirable for some.

Also for reference, this was my solution prior to the fix:
```
local function cursor_diagnostics()
    vim.diagnostic.open_float({
        scope = 'cursor',
        border = 'single',
    })
end
```

**Comment by atusy** [comment]
Yeah, such a conditional behavior requires desired tunings by users.
The basic idea would be below.
But, some would think they also want to jump when floating diagnostic is already open under the cursor).

```
if #require('lspsaga.diagnostic'):get_diagnostic({ cursor = true }) > 0 then
  require('lspsaga.diagnostic.show'):show_diagnostics({ cursor = true })
else
  require('lspsaga.diagnostic'):goto_next()
end
```
